tgbot/config.py

```diff
@@ -167,27 +145,37 @@ class Config:
     """
 
     tg_bot: TgBot
-    misc: Miscellaneous
-    db: Optional[DbConfig] = None
-    redis: Optional[RedisConfig] = None
+    db: DbConfig
+    redis: RedisConfig
+    misc: Optional[Miscellaneous]
 
 
-def load_config(path: str = None) -> Config:
-    """
-    This function takes an optional file path as input and returns a Config object.
-    :param path: The path of env file from where to load the configuration variables.
-    It reads environment variables from a .env file if provided, else from the process environment.
-    :return: Config object with attributes set as per environment variables.
+def load_config(env_file: Optional[str] = None):
     """
+    Load configuration from a specified or default .env file.
+
+    This function initializes configuration objects for the application
+    based on the settings defined in a .env file. If no specific file is
+    provided via `env_file`, it defaults to 'env'.
 
-    # Create an Env object.
-    # The Env object will be used to read environment variables.
-    env = Env()
-    env.read_env(path)
+    Parameters:
+        env_file (str, optional): Path to the .env file to use. Defaults to 'env'.
 
-    return Config(
-        tg_bot=TgBot.from_env(env),
-        # db=DbConfig.from_env(env),
-        # redis=RedisConfig.from_env(env),
-        misc=Miscellaneous(),
+    Returns:
+        Config: Config object containing settings loaded from the .env file.
+
+    Raises:
+        ValidationError: If any environment variables fail validation checks.
+    """
+    # Set the default .env file if none specified
+    if env_file is None:
+        env_file = ".env"
+
+    # Create instances with the specified or default .env file
+    config = Config(
+        tg_bot=TgBot(_env_file=env_file),
+        db=DbConfig(_env_file=env_file),
+        redis=RedisConfig(_env_file=env_file),
+        misc=Miscellaneous(_env_file=env_file),
```

Comment:
This could likely be refactored. If `env_file` is specified, we could invoke `set_env_file`. Since it's a class method, it should be able to alter the `.env` file path consistently across all subclasses. Check whether this is correct.

Reply:
do you mean something like this?
```code
class BaseSettings(_BaseSettings):
    model_config = SettingsConfigDict(
        extra="ignore", env_file=".env", env_file_encoding="utf-8", case_sensitive=False
    )

    @classmethod
    def set_env_file(cls, env_file_path: str):
        cls.model_config["env_file"] = env_file_path

    # Constructor that automatically sets the env file
    def __init__(self, *args, **kwargs):
        env_file = kwargs.pop('_env_file', None)
        if env_file:
            self.set_env_file(env_file)
        super().__init__(*args, **kwargs)
```

Reply:
@Latand